hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -82,14 +84,18 @@
   static String LOCATE_PREFETCH_LIMIT = "hbase.client.locate.prefetch.limit";
 
   private static final int DEFAULT_LOCATE_PREFETCH_LIMIT = 10;
+  private static final long GET_META_LOCATIONS_TIMEOUT_IN_MILLISECONDS = 2000; // 2 seconds
```

Comment:
I mean why we have a special hard coded time out here? Why not follow he default time out config? Or you want to introduce a new config?

Reply:
No, a new config is actually something I tried to avoid. "hbase.rpc.read.timeout"?  The default value is 1 min, a bit longer but I think it is ok, can adjust based on running results. If this is not the config you meant, please let me the know the config, thanks.

Reply:
I will go with this configuration and upload a new patch.

Reply:
A new patch takes "hbase.rpc.read.timeout". If there is a better config, please provide more details.